### DISCUSSION TITLE
AttributeError: 'ChromeOptions' has no attribute 'headless'

### DISCUSSION THREAD

**Discussion by benfield97** [ultrafunkamsterdam/undetected-chromedriver#1582] [General]
Really stumped on this issue, would love some help. I've wiped my entire computer and reinstalled everything, but still no luck. 

I have no code that tries to call headless.

chrome_options = uc.ChromeOptions()
chrome_options.add_argument("--remote-debugging-port=9222")
chrome_options.add_argument('--user-data-dir=/Users/user/Documents/chrome debug test')
chrome_options.binary_location = "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
bf_options = Options()

**Comment by vinq179** [comment]
options = ChromeOptions()
options.headless = True #add this line

**Reply by robertr2112** [reply]
I am also getting this error while running on heroku, however It works locally. I tried your recommended fix but I am still getting an error, albeit a different error.

I'm hoping you can offer another suggested fix.  
Thanks!

Here is the first code I tried:
   
    driver = uc.Chrome(headless=True,use_subprocess=False)
    path = f"https://www.nfl.com/schedules/{year}/REG{week}"
    driver.get(path)

With this I get the "AttributeError: 'ChromeOptions' object has no attribute 'headless'" error.

I tried your suggestion above and changed the code to try the following:

    uc_options = uc.ChromeOptions()

    # tried this first
    uc_options.headless = True 

    # Then tried this option
    uc_options.add_argument('--headless=new') 

    driver = uc.Chrome(options=uc_options)
    path = f"https://www.nfl.com/schedules/{year}/REG{week}" 
    driver.get(path)

Both options fail with the following traceback:

    Traceback (most recent call last):
      File "/app/lib/assets/python/nfl-scraper.py", line 147, in <module>
        driver.get(path)
      File "/app/.heroku/python/lib/python3.11/site-packages/undetected_chromedriver/__init__.py", line 629, in get_wrapped
        return orig_get(*args, **kwargs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/app/.heroku/python/lib/python3.11/site-packages/undetected_chromedriver/__init__.py", line 665, in get
        return super().get(url)
                 ^^^^^^^^^^^^^^^^
      File "/app/.heroku/python/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 353, in get
        self.execute(Command.GET, {"url": url})
      File "/app/.heroku/python/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 344, in execute
        self.error_handler.check_response(response)
      File "/app/.heroku/python/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py", line 229, in check_response
        raise exception_class(message, screen, stacktrace)
    selenium.common.exceptions.WebDriverException: Message: unknown error: session deleted because of page crash
    from tab crashed
      (Session info: chrome=117.0.5938.92)
    Stacktrace:
    #0 0x55e6fb18e6b3 <unknown>
    #1 0x55e6fae64036 <unknown>
    #2 0x55e6fae4d83b <unknown>
    #3 0x55e6fae4d081 <unknown>
    #4 0x55e6fae4c06c <unknown>
    #5 0x55e6fae4bfde <unknown>
    #6 0x55e6fae4b9c8 <unknown>
    #7 0x55e6fae6eb59 <unknown>
    #8 0x55e6faee651c <unknown>
    #9 0x55e6faecce93 <unknown>
    #10 0x55e6fae9f934 <unknown>
    #11 0x55e6faea071e <unknown>
    #12 0x55e6fb153cb8 <unknown>
    #13 0x55e6fb157bf0 <unknown>
    #14 0x55e6fb16219c <unknown>
    #15 0x55e6fb158808 <unknown>
    #16 0x55e6fb12527f <unknown>
    #17 0x55e6fb17ce88 <unknown>
    #18 0x55e6fb17d059 <unknown>
    #19 0x55e6fb18d843 <unknown>
    #20 0x7f98a50a7609 start_thread

**Reply by 724899020** [reply]
This setting is effective.

**Reply by D313373** [reply]
> options = ChromeOptions() options.headless = True #add this line

worked for me except I set it to False. strange that one should need to specify not to use a feature that isn't the default. But its working now and I am happy. Thank you!

**Comment by benfield97** [comment]
after downgrading selenium to 4.9.

I now get this errpr.

  File "/Users/macmini/Library/Python/3.9/lib/python/site-packages/undetected_chromedriver/__init__.py", line 466, in __init__
    super(Chrome, self).__init__(
  File "/Users/macmini/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/chrome/webdriver.py", line 84, in __init__
    super().__init__(
  File "/Users/macmini/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/chromium/webdriver.py", line 101, in __init__
    self.service.start()
  File "/Users/macmini/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/common/service.py", line 96, in start
    self._start_process(self._path)
  File "/Users/macmini/Library/Python/3.9/lib/python/site-packages/selenium/webdriver/common/service.py", line 197, in _start_process
    self.process = subprocess.Popen(
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/subprocess.py", line 1821, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 86] Bad CPU type in executable: '/Users/macmini/Library/Application Support/undetected_chromedriver/undetected_chromedriver'


I'm on an M1 mac mini

**Reply by robertr2112** [reply]
I got past the AttributeError: 'ChromeOptions' has no attribute 'headless' error with the following code:

    uc_options = uc.ChromeOptions()
    uc_options.add_argument('--headless')
    driver = uc.Chrome(options=uc_options, use_subprocess=False)

Unfortunately, I ran into other errors but this code seemed to fix the ChromeOptions error.

**Reply by sapfire65** [reply]
Exactly the same problem! Did you manage to decide?

**Reply by robertr2112** [reply]
I got past my other issues by adding the following options. 

       uc_options.add_argument('--no-sandbox')
     uc_options.add_argument('--disable-dev-shm-usage')

**Reply by sapfire65** [reply]
from selenium webdriver chrome?

**Reply by robertr2112** [reply]
From undetected-chromedriver with selenium

    import undetected_chromedriver as uc

But it would be the same for any webdriver-chrome used with selenium

**Reply by sapfire65** [reply]
`import undetected_chromedriver as uc
from undetected_chromedriver import ChromeOptions


uc_options = ChromeOptions()
uc_options.add_argument('--headless')
uc_options.add_argument('--no-sandbox')
uc_options.add_argument('--disable-dev-shm-usage')
driver = uc.Chrome(options=uc_options, use_subprocess=False)

driver.get('https://nowsecure.nl')
driver.save_screenshot('now-secure.png')`

does not work... Is there something that works?

**Comment by Zemichal** [comment]
If you're not trying to use headless try putting this before your driver: chrome_options.headless = False

**Reply by davidwitten** [reply]
+1, I just do:

```
chrome_options = Options()
chrome_options.headless = False
dr = uc.Chrome(options=chrome_options)
```
and it works perfectly

**Reply by Namemi2002** [reply]
Thanks, it works
